src/bootstrap/index.ts

```diff
@@ -74,7 +74,7 @@ async function retry_until_rc(cmd: string, expected_rc=0, maxRetries=12, timeout
 async function microk8s_init(addons) {
     // microk8s needs some additional things done to ensure it's ready for Juju.
     // Add the given addons if any were given.
-    await exec_as_microk8s("microk8s status --wait-ready");
+    await exec_as_microk8s("microk8s status --wait-ready --timeout 900");
```

Comment:
shouldn't this fail if it timesout?

```suggestion
    if (! await exec_as_microk8s("microk8s status --wait-ready --timeout 900")){
        core.setFailed("Timed out waiting for microk8s to be ready")
        return false;
    }
```

Reply:
@addyess 
from what I have seen `microk8s status --wait-ready --timeout 900` will not return non-zero if times out. At least my CI was kept running

or ?

Reply:
so, we're just hoping microk8s is ready after 15 min?  
@neoaggelos any thought on whether this `microk8s` command should error on timeout?

Reply:
@berkayoz do you have any thoughts on a situation where 15min wait wouldn't finish

Reply:
The `--wait-ready` [waits](https://github.com/canonical/microk8s/blob/f4eaff59f1584f11fceb184d93dedca80de712d5/scripts/wrappers/common/utils.py#L90-L96) until the default `kubernetes` service shows up and the nodes get into the `Ready` state. I wouldn't expect too many issues around snap services like `kubelet` and such, the most common would be image pull related problems since calico needs to be up inorder for the node to be `Ready`.

It also seems like the `microk8s status --wait-ready` command doesn't return a failure exit code, it might be a good idea to add one incase of a timeout.

Reply:
added an issue on microk8s https://github.com/canonical/microk8s/issues/3927